packages/thirdweb/src/react/web/ui/prebuilt/NFTProvider/index.tsx

```diff
@@ -0,0 +1,94 @@
+"use client";
+
+import { createContext, useContext } from "react";
+import { useReadContract } from "src/exports/react.native.js";
+import { isERC721 } from "src/extensions/erc721/read/isERC721.js";
+import { totalSupply } from "src/extensions/erc1155/__generated__/IERC1155/read/totalSupply.js";
+import { isERC1155 } from "src/extensions/erc1155/read/isERC1155.js";
+import type { Chain } from "../../../../../chains/types.js";
+import type { ThirdwebClient } from "../../../../../client/client.js";
+import {
+  type ThirdwebContract,
+  getContract,
+} from "../../../../../contract/contract.js";
+import { useNFTMedia } from "./useNFTMedia.js";
+
+export type NFTProviderSetup = {
+  contract: ThirdwebContract;
+  isLoading: boolean;
+  /**
+   * The media URI of the NFT
+   */
+  media?: string;
+  contractType?: "ERC721" | "ERC1155";
+  /**
+   * The total supply of the token. Only aplpicable to ERC1155
+   * Defaults to 1n for ERC721
+   */
+  supply?: bigint;
+};
+
+type NFTProviderProps = {
+  contractAddress: string;
+  chain: Chain;
+  client: ThirdwebClient;
+  tokenId: bigint;
+  overrideMediaField?: string;
+};
+
+export const NFTProviderContext = /* @__PURE__ */ createContext<
+  NFTProviderSetup | undefined
+>(undefined);
+
+export function NFTProvider(props: React.PropsWithChildren<NFTProviderProps>) {
```

Comment:
I wonder if we just do:

`NFT`
and then `NFT.Media`

meaning you could build it like this:

```jsx
import {NFT} from "thirdweb/react";

<NFT contract={...} tokenId={0n}>
   <NFT.Media className={...} />
</NFT>
````

Reply:
@jnsdls How does this affect treeshaking ser

Reply:
I'm assuming this is the pattern you are talking about:
```tsx
export function NFT(props: React.PropsWithChildren<NFTProviderProps>) {
  return (
    <NFTProviderContext.Provider value={props}>
      {props.children}
    </NFTProviderContext.Provider>
  );
}

NFT,Media = (...) => {
 return <>...</>
}
```

Reply:
yeah, they *usually* will be used in combination so I *think* it's fine? Maybe we even enforce that they are used in combination (have to wrap media in provider first?)

Reply:
I love it actually. NFT.Media, NFT.Description etc. Very ez to remember

Reply:
yep exactly